pex/environment.py

```diff
@@ -221,7 +221,11 @@ def can_add(self, dist):
 
     # Wheel filename format: https://www.python.org/dev/peps/pep-0427/#file-name-convention
     # `{distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl`
-    wheel_tags = '-'.join(filename.split('-')[-3:])  # `{python tag}-{abi tag}-{platform tag}`
+    wheel_components = filename.split('-')
+    if len(wheel_components) < 3:
+      return False
+
+    wheel_tags = '-'.join(wheel_components[-3:])  # `{python tag}-{abi tag}-{platform tag}`
     if self._supported_tags.isdisjoint(tags.parse_tag(wheel_tags)):
```

Comment:
Would have gone boom! here in `tags.parse_tags`:
https://github.com/pantsbuild/pex/blob/2a8b950284d4a8be5eb2bcf9dcb63e2bc5f60fe4/pex/vendor/_vendored/packaging/packaging/tags.py#L73-L75